devops/providers/azure-devops/templates/infrastructure/README.md

```diff
@@ -0,0 +1,204 @@
+
+## Prerequisites
+
+- An provisioned Azure DevOps instance
+- Service principal
+- Storage account for tracking remote terraform state. We recommend [backing](https://azure.microsoft.com/en-us/services/backup/) this storage account up in Azure. Terraform will delete all provisioned resources if the remote state file is either corrupted or deleted.
+- Key Vault resource including the service principal and storage account credentials following these [instructions](../../README.md#required-key-vault-secrets).
+- A git repo containing the terraform deployment template(ie Github, Github Enterprise, Azure DevOPS git, etc).
+- Azure DevOPS Github integration installed on the upstream repo
+
+## Build Pipeline
+
+### Description
+
+The goal for this particular pipeline is to download the target cobalt template manifest artifacts onto the build agent and apply the terraform execution plan onto the Dev Integration environment.
+
+The build will reconcile the current environment cloud resource state with the desired resource state to determine which resources to add, remove or modify. Terraform uses a remote storage account residing in each Azure environment to maintain latest state.
+
+Once the resources have been provisioned within the Dev Int Environment, the test harness will run the automated unit and integration tests to validate the functionality end-to-end.
+
+The test harness will automatically tear down all provisioned resources to help minimize azure costs and duplicate cloud objects.
+
+### Build Trigger
+
+- Cobalt is a library of infrastructure manifests represented as Terraform templates(aka IAC: Infrastructure as Code).
+
+- Each template defines multiple composable building-block terraform modules and assembles them together to produce a larger system.
+
+- This pipeline's deployment process is built around a repo that holds the template to provision.
+
+- Operational changes are made to the running system by making commits on this repo.
+
+- Follow these [instructions](https://docs.microsoft.com/en-us/azure/devops/pipelines/repos/github?view=azure-devops&tabs=yaml#private-github-repository-or-azure-pipelines-in-a-private-project) to link the IAC repo to the build pipeline.
+
+- Here's some more [background](https://docs.microsoft.com/en-us/azure/devops/pipelines/repos/github?view=azure-devops&tabs=yaml#ci-triggers) on how CI triggers work in Azure DevOPS. 
+
+- The build pipeline is triggered on new commits made to any feature branch.
+
+#### Getting the source code
+
+The linked IAC git repo will be cloned and pulled onto the build agent machine. Here's some more [info](https://docs.microsoft.com/en-us/azure/devops/pipelines/repos/github?view=azure-devops&tabs=yaml#getting-the-source-code) on how that will operate from the perspective of the Azure unified pipeline.
+
+#### Git: Master Branch Policies
```

Comment:
Supposing that consumers of this may not use Git it would be helpful to have a description of the policies along with their benefits in a way that is not specific to Git. Thoughts?

Reply:
This pipeline will only support git-based repositories. I'd imagine a separate pipeline definition for non-git source repos.

Reply:
+1 on being more explicit with the recommended policies.